### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by KIsmoilov**
**The following features have been implemented below:**

* Removed current_user method in ApplicationController, devise will provide us with one.
* [Installed and setup devise](https://github.com/heartcombo/devise#getting-started).

> 1. Can register a new user.
> 2. User logs in with a combination of email and password.
> 3. Hashed password can be stored in the database.
> 4. Asks for confirmation of email.
> 5. Can reset password.

 * [Modifed the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

Screen Shot 2023-03-09 at 11 31 03 AM

Screen Shot 2023-03-09 at 11 31 09 AM

Screen Shot 2023-03-09 at 11 31 24 AM

Screen Shot 2023-03-09 at 11 31 28 AM

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```
Comment (julie-ify) [inline app/controllers/application_controller.rb:4]: - [ ] Despite installing devise I am still able to visit several supposedly unauthorized routes, like the new post route. Kindly ensure that only logged-in users can use the app by adding the code below to this file:

  ```
  before_action :authenticate_user!
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @KIsmoilov and @ndikumanaisaie,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KIsmoilov** [comment]
**Implemented the required changes.**

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @KIsmoilov 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️ 
- Devise well added✔️ 
- Registering new user ✔️ 
- No linter errors✔️ 


--[optional] please kindly remove the live demo section in your readme since you don't yet have a deployed version of your app. this will keep your readme clean. you can add it when you have one. nice job

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**